jupyter_server/gateway/connections.py

```diff
@@ -30,6 +30,10 @@ class GatewayWebSocketConnection(BaseKernelWebsocketConnection):
 
     retry = Int(0)
 
+    # When opening ws connection to gateway, server already negotiated subprotocol with notebook client.
+    # Same protocol must be used for client and gateway, so legacy ws subprotocol for client is enforced here.
+    kernel_ws_protocol = Unicode("")
```

Comment:
```suggestion

    kernel_ws_protocol = Unicode(
        "",
        allow_none=True,
        config=True
    )

```

Reply:
I _think_ we want to keep this configurable, since in principle, someone could configure both jupyter-server and the gateway to use the `v1` protocol.